### DISCUSSION TITLE
Naming conventions, make_ vs new_ in Vlib

### DISCUSSION THREAD

**Discussion by eptx** [vlang/v#16508] [Questions and Answers]
Looking through https://modules.vlang.io/ there are a number of functions than begin with **make_** rather than **new_**. Is this an inconsistency to be cleaned up or is there a distinction when to use one or the other?

**Comment by spytheman** [comment]
The `new_` prefix is the preferred one. The `make_` functions are either not public, are inside tests, or are wrappers to C functions (where similarity to the underlying C library and thus its documentation is more valuable than the V convention of using `new_`).

**Comment by spytheman** [comment]
You can compare `git grep 'pub fn make_' | wc -l` = 8, vs `git grep 'pub fn new_' | wc -l` = 138 .
For the remaining public make_ functions:
```
vlib/io/readerwriter.v:27:pub fn make_readerwriter(r Reader, w Writer) ReaderWriterImpl {
vlib/sokol/gfx/gfx.c.v:33:pub fn make_buffer(desc &BufferDesc) Buffer {
vlib/sokol/gfx/gfx.c.v:38:pub fn make_image(desc &ImageDesc) Image {
vlib/sokol/gfx/gfx.c.v:43:pub fn make_shader(desc &ShaderDesc) Shader {
vlib/sokol/gfx/gfx.c.v:48:pub fn make_pipeline(desc &PipelineDesc) Pipeline {
vlib/sokol/gfx/gfx.c.v:53:pub fn make_pass(desc &PassDesc) Pass {
vlib/sokol/sgl/sgl.c.v:43:pub fn make_context(desc &ContextDesc) Context {
vlib/sokol/sgl/sgl.c.v:69:pub fn make_pipeline(desc &gfx.PipelineDesc) Pipeline {
```

I am not sure about the first one, it may be renamed to new_ without much problems, with may be a small deprecation for a while.

The others are in the sokol wrappers, and are named like that, to be more similar to the sokol names (which do use make_ as a prefix), but could be renamed also with a small deprecation period.

**Comment by medvednikov** [comment]
I agree, all public functions starting with make_ should be renamed.

**Reply by medvednikov** [reply]
On the other hand, if it's just sokol wrappers, it's not a big issue.
